### PR TITLE
change web search

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/web/issues/432

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/SearchService.java [hunk 1]
```diff
@@ -98,16 +106,18 @@ public List<LatLon> getBboxCoords(List<String> coords) {
...
     }
     
-    public List<SearchResult> search(double lat, double lon, String text) throws IOException {
+    public List<Feature> search(double lat, double lon, String text, String locale) throws IOException, XmlPullParserException {
         if (!osmAndMapsService.validateAndInitConfig()) {
             return Collections.emptyList();
         }
-        SearchUICore searchUICore = new SearchUICore(MapPoiTypes.getDefault(), SEARCH_LOCALE, false);
+        SearchUICore searchUICore = new SearchUICore(getMapPoiTypes(locale), locale, false);
+        searchUICore.setTotalLimit(1000);
```
Comment (RZR-UA) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/SearchService.java:114]: please avoid magic values

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/SearchService.java [hunk 2]
```diff
@@ -361,6 +383,16 @@ public Map<String, List<String>> searchPoiCategories() {
         return res;
     }
     
+    private MapPoiTypes getMapPoiTypes(String locale) throws XmlPullParserException, IOException {
+        MapPoiTypes mapPoiTypes = MapPoiTypes.getDefault();
+        String localPath = locale.equals("en") ? "values" : "values-" + locale;
+        Map<String, String> phrases = Algorithms.parseStringsXml(new File(andTranslationsLocation + localPath + "/phrases.xml"));
```
Comment (RZR-UA) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/SearchService.java:389]: dangerous File access (externally-defined by locale string). should be sanitized. at minimum, remove '/' chars from `locale`

File: java-tools/OsmAndServer/src/main/resources/application.yml [hunk 3]
```diff
@@ -30,6 +30,7 @@ osmand:
   srtm.location: ${SRTM_LOCATION:}
   weather.location: ${WEATHER_LOCATION:}
   wiki.location: ${osmand.files.location}/wikidata/wikidata_commons_osm.sqlitedb
+  android.translations.location: ${osmand.files.location}/android/OsmAnd/res/
```
Comment (RZR-UA) [inline java-tools/OsmAndServer/src/main/resources/application.yml:33]: osmand.files.location: /var/www-download

there is no `/android/OsmAnd/res/` in the specified directory

please think another way to get the resources and refresh them regularly

possible path is workspace of Jenkins WebSite_UpdateFrontend job (depened to Web updates...)

**Review by RZR-UA — CHANGES_REQUESTED** [review]

**Review by RZR-UA — APPROVED** [review]

**Comment by RZR-UA** [comment]
- [ ] https://test.osmand.net/routing/search/get-poi-address?lat=48.17473&lon=24.437403 fix 400 "Error get poi address!" (error is reported in browser)

**Comment by RZR-UA** [comment]
- [ ] fix divide by zero error

```
curl 'https://test.osmand.net/routing/search/search-poi?locale=en' -X POST -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:129.0) Gecko/20100101 Firefox/129.0' -H 'Accept: */*' -H 'Accept-Language: en-US,en;q=0.5' -H 'Accept-Encoding: gzip, deflate, br, zstd' -H 'Referer: https://test.osmand.net/map/search/' -H 'Content-Type: application/json' -H 'Origin: https://test.osmand.net' -H 'Connection: keep-alive' -H 'Cookie: email=youonly%40ukr.net; XSRF-TOKEN=c4f2ae4f-7e7d-4dc7-90a6-71d6d072f80d; JSESSIONID=2B920FB7DF0ADAC56D8D8BB5252B9087' -H 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: no-cors' -H 'Sec-Fetch-Site: same-origin' -H 'Priority: u=4' -H 'Pragma: no-cache' -H 'Cache-Control: no-cache' --data-raw '{"categories":[],"northWest":"48.237308654125236,24.481573104858402","southEast":"48.182684785134384,24.618902206420902","savedNorthWest":"48.23565029755308,24.29386138916016","savedSouthEast":"48.12634074841738,24.568519592285156","prevCategoriesCount":null}'
```

```
java.lang.ArithmeticException: / by zero
        at net.osmand.server.api.services.SearchService.searchPoi(SearchService.java:164)
        at net.osmand.server.controllers.pub.SearchController.searchPoi(SearchController.java:67)
        at jdk.internal.reflect.GeneratedMethodAccessor67.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
```

**Review by RZR-UA — APPROVED** [review]

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/OsmAndMapsService.java [hunk 4]
```diff
@@ -231,92 +231,94 @@ public double importance() {
 	public class BinaryMapIndexReaderReference {
 		File file;
 		private static final int WAIT_LOCK_CHECK = 10;
-		ConcurrentHashMap<BinaryMapIndexReader, Boolean> readers = new ConcurrentHashMap<>();
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/OsmAndMapsService.java:234]: too complicated we should reuse readers for all locales

**Review by ivanPyrohivskyi — APPROVED** [review]

**Pull request merged**